syntax = "proto3";

enum SelectOption {
    GET = 0;
    GET_ALL = 1;
    ADD = 2;
    REGISTER=3;
    LOGIN=4;
    CURRENT_USER=5;
    REFRESH_TOKEN=6;
    CREATE = 7;
    UPDATE =8;
    REDUCE = 9;
    DELETE =10;
    PAYMENT_DONE = 11;
}


message Payment {
    int32 id =1;
    string payment_id=2;
    string order_id =3;
    string user_id=4;
    float amount = 5;
    int64 card_number = 6;
    int32 exp_month = 7;
    int32 exp_year = 8;
    int32 cvc = 9;
    PaymentStatus payment_status = 10;
    SelectOption option =11;
    int64 http_status_code = 12;
    string error_message = 13;

}



message User {
    int32 id =1;
    string user_id=2;
    string username =3;
    string email=4;
    string password =5;
    string access_token = 6;
    string refresh_token = 7;
    string error_message = 8;
    SelectOption option =9;
    int64 http_status_code = 10;
}

enum OrderStatus {
  IN_PROGRESS = 0;
  COMPLETED = 1;
}

enum PaymentStatus {
    PENDING = 0;
    PAID = 1;
    FAILED = 2;
}

message Order {
    int32 id =1;
    string order_id=2;
    string product_id =3;
    string user_id = 4;
    string username =5;
    string email=6;
    int64 quantity= 7;
    string shipping_address = 8;
    string customer_notes = 9;
    int64 stock_level = 10;
    bool is_product_available = 11;
    bool is_stock_available  = 12;
    string message = 13;
    string error_message = 14;
    OrderStatus order_status = 15;
    PaymentStatus payment_status = 16;
    SelectOption option =17;
    int64 http_status_code = 18;
}

message OrderList {
    repeated Order orders = 1;
}